packages/node/src/integrations/localvariables/localvariables-async.ts

```diff
@@ -0,0 +1,287 @@
+  const properties = await session.post('Runtime.getProperties', { objectId, ownProperties: true });
+
+  return {
+    [name]: properties.result
+      .filter(v => v.name !== 'length' && !isNaN(parseInt(v.name, 10)))
+      .sort((a, b) => parseInt(a.name, 10) - parseInt(b.name, 10))
+      .map(v => v?.value?.value),
+  };
+}
+
+async function unrollObject(session: Session, objectId: string, name: string): Promise<Variables> {
+  const properties = await session.post('Runtime.getProperties', { objectId, ownProperties: true });
+
+  return {
+    [name]: properties.result
+      .map<[string, unknown]>(v => [v.name, v?.value?.value])
+      .reduce((obj, [key, val]) => {
+        obj[key] = val;
+        return obj;
+      }, {} as Variables),
+  };
+}
+
+function unrollOther(prop: Runtime.PropertyDescriptor): Variables {
+  if (prop?.value?.value) {
+    return {
+      [prop.name]: prop.value.value,
+    };
+  } else if (prop?.value?.description && prop?.value?.type !== 'function') {
+    return {
+      [prop.name]: `<${prop.value.description}>`,
+    };
+  }
+
+  return {};
+}
+
+async function getLocalVariables(session: Session, objectId: string): Promise<Variables> {
+  const properties = await session.post('Runtime.getProperties', { objectId, ownProperties: true });
+  let variables = {};
+
+  for (const prop of properties.result) {
+    if (prop?.value?.objectId && prop?.value.className === 'Array') {
+      const id = prop.value.objectId;
+      variables = { ...variables, ...(await unrollArray(session, id, prop.name)) };
+    } else if (prop?.value?.objectId && prop?.value?.className === 'Object') {
+      const id = prop.value.objectId;
+      variables = { ...variables, ...(await unrollObject(session, id, prop.name)) };
+    } else if (prop?.value?.value || prop?.value?.description) {
+      variables = { ...variables, ...unrollOther(prop) };
+    }
+  }
+
+  return variables;
+}
+
+/**
+ * Adds local variables to exception frames
+ *
+ * Default: 50
+ */
+export class LocalVariablesAsync implements Integration {
+  public static id: string = 'LocalVariablesAsync';
+
+  public readonly name: string = LocalVariablesAsync.id;
+
+  private readonly _cachedFrames: LRUMap<string, FrameVariables[]> = new LRUMap(20);
+  private _rateLimiter: RateLimitIncrement | undefined;
+  private _shouldProcessEvent = false;
+
+  public constructor(private readonly _options: Options = {}) {}
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_addGlobalEventProcessor: (callback: EventProcessor) => void, _getCurrentHub: () => Hub): void {
+    // noop
+  }
+
+  /** @inheritdoc */
+  public setup(client: NodeClient): void {
+    const clientOptions = client.getOptions();
+
+    if (!clientOptions.includeLocalVariables) {
+      return;
+    }
+
+    import('node:inspector/promises')
+      .then(async inspector => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const { Session } = inspector as any;
```

Comment:
Can we use the `node:inspector/promises` type in here?

Reply:
Yeah, I think the `Session` type def an go in the definition file and it'll fix all this